src/Rules/Install.hs

```diff
@@ -0,0 +1,299 @@
+getLibExecDir = (-/- "bin") <$> installGhcLibDir
+
+-- | install script executables to @prefix/lib/bin@
+installLibExecScripts :: Action ()
+installLibExecScripts = do
+  need libExecScripts
+  libExecDir <- getLibExecDir
+  dest <- destDir
+  installDir (dest ++ libExecDir)
+  forM_ libExecScripts $ \script -> do
+    installScript script (dest ++ libExecDir)
+
+-- TODO: ghci
+-- driver/ghci/ghc.mk:
+-- INSTALL_BINS += driver/ghci/dist/build/tmp/$(driver/ghci_dist_PROG_VER)
+
+-- | install binary executables to @prefix/lib/bin@
+installLibExecs :: Action ()
+installLibExecs = do
+  libExecDir <- getLibExecDir
+  dest <- destDir
+  installDir (dest ++ libExecDir)
+  forM_ installPkgs $ \pkg -> do
+      stg' <- latestBuildStage pkg
+      case stg' of
+          Nothing -> return ()
+          Just stg -> do
+              let context = programContext stg pkg
+              let bin = inplaceLibBinPath -/- programName context <.> exe
+              need [bin]
+              installProgram bin (dest ++ libExecDir)
+              when (pkg == ghc) $ do
+                  moveFile (dest ++ libExecDir -/- programName context <.> exe)
+                           (dest ++ libExecDir -/- "ghc" <.> exe)
+
+-- TODO: Consider Stage1Only
+installPkgs :: [Package]
+installPkgs =
+  [ ghc, ghcPkg, ghcSplit, hp2ps
+  , hpc, hsc2hs, runGhc, unlit ]
+
+-- | install wrapper script executables to @prefix/bin@
+installBins :: Action ()
+installBins = do
+  binDir <- setting InstallBinDir
+  dest <- destDir
+  installDir (dest ++ binDir)
+  forM_ installPkgs $ \pkg ->
+    withLatestBuildStage pkg $ \stg -> do
+        let context = programContext stg pkg
+        version <- setting ProjectVersion
+        let binName = if pkg == ghc then "ghc-" ++ version <.> exe
+                                    else programName context ++ "-" ++ version <.> exe
+        let symName = if pkg == ghc then "ghc" <.> exe
+                                    else programName context <.> exe
+        case lookup context installWrappers of
+            Nothing -> return ()
+            Just wrapper -> do
+                 libDir <- installGhcLibDir
+                 contents <- interpretInContext context $
+                                wrapper
+                                (WrappedBinary (dest ++ libDir) symName)
+                 withTempDir $ \tmp -> do
+                     let tmpfile = tmp -/- binName
+                     writeFileChanged tmpfile contents
+                     installProgram tmpfile (dest ++ binDir)
+                 win <- windowsHost
+                 when (not win) $
+                     linkSymbolic (dest ++ binDir -/- binName)
+                                  (dest ++ binDir -/- symName)
+
+getLibraryFile :: Context -> String
+getLibraryFile context@Context{..} =
+  let path      = buildPath context
+      libPrefix = path -/- "libHS" ++ pkgNameString package
+  in  libPrefix ++ waySuffix way <.> "a"
+
+stripCmd :: Context -> Action String
```

Comment:
This function is used in one spot only -- should it be made local? What does return value `:` mean?